nix/vast/default.nix

```diff
@@ -102,10 +101,12 @@ stdenv.mkDerivation rec {
     "-DVAST_ENABLE_STATIC_EXECUTABLE:BOOL=ON"
     "-DCMAKE_INTERPROCEDURAL_OPTIMIZATION:BOOL=ON"
   ] ++ lib.optional disableTests "-DVAST_ENABLE_UNIT_TESTS=OFF"
-    # Plugin Section
-    ++ lib.optional (withPlugins != [])
-       "-DVAST_PLUGINS=${lib.concatStringsSep ";" withPlugins}"
-    ++ extraCmakeFlags;
+  # Plugin Section
+  ++ lib.optionals (withPlugins != []) [#Exp: ["broker" "pcap"]
+    "-DVAST_PLUGINS=${lib.concatImapStringsSep ";" (pos: x: "plugins/" + x ) withPlugins}"
+    "-DCMAKE_INSTALL_LIBDIR=lib"
```

Comment:
`lib` should be the default value for `CMAKE_INSTALL_LIBDIR`. Why would you need to specify this manually here?

Reply:
`CMAKE_INSTALL_LIBDIR` missing from the Non-static build option of the withPlugins.

Reply:
> @tobim is on holiday this week, so this PR will likely sit until early next week. He's the guy doing all of our Nix stuff, and it's probably best if we let him review this.

Thanks, I got it. So I did a minimal patch to here which just modified args of withPlugin.
For the Non-static build with Nix,  it might still have a problem with `withPlugin`.  Anyway, I have synced the build to the static.

Reply:
You only hit that codepath if CMAKE_INSTALL_LIBDIR is not a subdirectory of CMAKE_INSTALL_PREFIX, which it always should be.

Reply:
https://github.com/tenzir/vast/pull/1786 should fix the core issue, so this workaround will hopefully not be necessary.

Reply:
I don't need specific `CMAKE_INSTALL_LIBDIR` anymore. Just ENABLE_STATIC_PLUGIN in default build. https://github.com/tenzir/vast/pull/1769/commits/9069f3b7b5ae10f3e15daf9ac11a4b42b80ef7ec#diff-20342b514d1c1f399318d01baf84c6982f37b7aa789a10a9b944127068e5398eR106